invenio_rdm_records/resources/serializers/schemas.py

```diff
@@ -79,3 +79,100 @@ def get_contributors(self, obj):
         return [
             c["person_or_org"]["name"] for c in obj["metadata"].get("contributors", [])
         ] or missing
+
+
+class DumperMixin(Schema):
+    """A mixin that adds post-dump and pre-dump processing functionality to the `Schema` class.
+
+    This mixin provides a way to modify the serialized or deserialized output of a schema after or before it has been dumped.
+    """
+
+    def __init__(self, dumpers=None, **kwargs):
+        """Initialize a new instance of the `DumperMixin` class.
+
+        :param dumpers: A list of `DumpProcessorMixin` instances used to modify the serialized output, defaults to None.
+        :type dumpers: list, optional
+        :raises AssertionError: If any item in the `dumpers` list is not an instance of `DumpProcessorMixin`.
+        """
+        super().__init__(**kwargs)
+        self.dumpers = dumpers or []
+        assert all(isinstance(p, FieldDumper) for p in self.dumpers)
```

Comment:
I've already commented on this and is not clear to me how do we want to proceed. This is parameter field checking, not business logic. What was the consensus? I'd like someone to tell me if I should stop bringing this up 😇 

ref https://github.com/inveniosoftware/invenio-rdm-records/pull/1258#discussion_r1143114019

cc @kpsherva @zzacharo  this might be worth a discussion

Reply:
I keeped it after that discussion. As @ntarocco mentioned, that's an unrecoverable error for the system.

I am not a big fan of asserts in the code either but seemed ok in this case.

Reply:
Discussed IRL with Karolina and Nico. We should remove this checks.

While I do agree that it is _an unrecoverable error for the system_, it is an error due to the type/class of the object, not due to the business logic. It would be the same if we pass an object that is not a custom field in the config, or an object that is not a serializer (long etc.).

A different case is the one mentioned in the transformer/queryparser PR. There we need to make sure that a series of preconditions/business logic is fulfilled (i.e. two parameters that are dependant are provided). In those cases we should put an `assert`.